crates/rpc/rpc/src/debug.rs

```diff
@@ -403,17 +405,17 @@ where
 
         let mut inspector = TracingInspector::new(inspector_config);
 
-        let (res, inspector) = self
+        let (res, env, inspector) = self
             .inner
             .eth_api
             .spawn_with_call_at(call, at, overrides, move |db, env| {
-                let (res, _) = this.eth_api().inspect(db, env, &mut inspector)?;
-                Ok((res, inspector))
+                let (res, _) = this.eth_api().inspect(db, env.clone(), &mut inspector)?;
+                Ok((res, env, inspector))
```

Comment:
this clone is a bit redundant if we could return a copy of the env.tx.gas_limit instead

Reply:
@mattsse It seems like the clone is still needed.